src/applications/personalization/profile/components/notification-settings/NotificationRadioButtons.jsx

```diff
@@ -0,0 +1,266 @@
+    warningSpan = (
+      <span
+        className="rb-input-warning-message"
+        role="alert"
+        id={warningSpanId}
+      >
+        <i className="fas fa-exclamation-triangle vads-u-margin-x--1" />{' '}
+        <span className="sr-only">Warning</span> {warningMessage}
+      </span>
+    );
+  }
+
+  let successSpan = '';
+  let successSpanId = undefined;
+  if (successMessage) {
+    successSpanId = `${id}-success-message`;
+    successSpan = (
+      <span
+        className="rb-input-success-message"
+        role="alert"
+        id={successSpanId}
+      >
+        <i className="fas fa-check-circle vads-u-margin-x--1" />{' '}
+        <span className="sr-only">Success</span> {successMessage}
+      </span>
+    );
+  }
+
+  // Calculate required.
+  let requiredSpan = undefined;
+  if (required) {
+    requiredSpan = <span className="form-required-span">(*Required)</span>;
+  }
+
+  const buttonOptions = _.isArray(options) ? options : [];
+  const storedValue = value?.value;
+  const optionElements = buttonOptions.map((option, optionIndex) => {
+    let optionLabel;
+    let optionValue;
+    let optionAdditional;
+    if (_.isString(option)) {
+      optionLabel = option;
+      optionValue = option;
+    } else {
+      optionLabel = option.label;
+      optionValue = option.value;
+      if (option.additional) {
+        optionAdditional = <div>{option.additional}</div>;
+      }
+    }
+    const checked = optionValue === storedValue ? 'checked=true' : '';
+    const buttonAriaDescribedby =
+      (checked && ariaDescribedby?.[optionIndex]) || null;
+    return (
+      <div
+        key={optionAdditional ? undefined : optionIndex}
+        className="form-radio-buttons"
+      >
+        <div className="errorable-radio-button rb-buttons">
+          <input
+            checked={checked}
+            id={`${id}-${optionIndex}`}
+            name={name}
+            type="radio"
+            onMouseDown={onMouseDown}
+            onKeyDown={onKeyDown}
+            value={optionValue}
+            onChange={handleChange}
+            aria-describedby={buttonAriaDescribedby}
+          />
+
+          <label
+            name={`${name}-${optionIndex}-label`}
+            htmlFor={`${id}-${optionIndex}`}
+          >
+            {optionLabel}
+          </label>
+          {option.content}
+        </div>
+      </div>
+    );
+  });
+
+  const fieldsetClass = classNames('rb-fieldset-input', {
+    'rb-input-error': errorMessage,
+    'rb-input-warning': warningMessage,
+    'rb-input-success': successMessage,
+    [additionalFieldsetClass]: additionalFieldsetClass,
+  });
+
+  const legendClass = classNames('rb-input-notify-label', {
+    [additionalLegendClass]: additionalLegendClass,
+  });
+
+  return (
+    <fieldset className={fieldsetClass}>
+      <span className={legendClass}>
+        {label}
+        {requiredSpan}
+      </span>
+      {errorSpan}
+      {warningSpan}
+      {successSpan}

```

Comment:
It'd likely be best to ensure that only one of these special spans could exist at any one time, especially since there is only a single colored rule line on the left hand side of the component. But that's something to think about as we move forward.

Reply:
I was thinking that.